From 4b735fd0e15e1f03059b974dff1c01b16a7ae57b Mon Sep 17 00:00:00 2001
From: francisrj2 <55339121+teelrabbit@users.noreply.github.com>
Date: Sat, 16 Mar 2024 13:11:42 -0700
Subject: [PATCH] #issue-1846 fixed by updating imp to importlib.

---
 setup.py | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 459e6128a..c65f1fa04 100755
--- a/setup.py
+++ b/setup.py
@@ -5,15 +5,18 @@
 # Use of this source code is governed by a BSD-3-clause license that can be
 # found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
 
-import imp
+import importlib.util
 import os
 from setuptools import setup, find_packages
 
 # Get the coremltools version string
 coremltools_dir = os.path.join(os.path.dirname(__file__), "coremltools")
-version_module = imp.load_source(
-    "coremltools.version", os.path.join(coremltools_dir, "version.py")
-)
+version_file = os.path.join(coremltools_dir, "version.py")
+
+spec = importlib.util.spec_from_file_location("coremltools.version", version_file)
+version_module = importlib.util.module_from_spec(spec)
+spec.loader.exec_module(version_module)
+
 __version__ = version_module.__version__
 
 README = os.path.join(os.getcwd(), "README.md")
